src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -164,14 +162,13 @@ int replicationFactor() {
     public void run() {
       try {
         TopicState topicState = topicState();
-        if (topicState == null) {
-          if (!_topicCreationEnabled) {
-            throw new RuntimeException(_serviceName + ": topic, " + _topic + ", does not exist and topic creation is not enabled.");
-          }
-
-          topicState = createTopic();
+        
+        if (topicState == null && !_topicCreationEnabled) {
+          throw new RuntimeException(_serviceName + ": topic, " + _topic + ", does not exist and topic creation is not enabled.");
         }
 
+        topicState = createTopic();
```

Comment:
NPE can't happen because createTopic() returns a new topicState or else throws an exception if the topic was not created.  The code that you propose is still more clear so I will make that change.